Moving Windows security info to new page (#31851)
kubernetes/website: content/en/docs/concepts/security/windows-security.md

---
reviewers:
- jayunit100
- jsturtevant
- marosset
- perithompson
title:    Security For Windows Nodes
content_type: concept
weight: 75
---

<!-- overview -->

This page describes security considerations and best practices specific to the Windows operating system.

<!-- body -->

> 

Reviewer: ```suggestion
## Protection for Secret data on nodes
```

On Windows, data from Secrets are written out in clear text onto the node's local
storage (as compared to using tmpfs / in-memory filesystems on Linux). As a cluster
operator, you should take both of the following additional measures:

1. Use file ACLs to secure the Secrets' file location.
1. Apply volume-level encryption using [BitLocker](https://docs.microsoft.com/en-us/windows/security/information-protection/bitlocker/bitlocker-how-to-deploy-on-windows-server).

> 

Reviewer: ```suggestion

## Container isolation by user identity

```
?

Author: Hmm, specifying the username doesn't really impact isolation, let me think about what to use for the section header.

Author: I re-named this section to `Container Users` and added more details.

PTAL

Reviewer: Sounds good

[RunAsUsername](/docs/tasks/configure-pod-container/configure-runasusername)
can be specified for Windows Pods or containers to execute the container
processes as a node-default user. This is roughly equivalent to
[RunAsUser](/docs/concepts/policy/pod-security-policy/#users-and-groups).

> 

Reviewer: ```suggestion

## Pod-level security isolation

```

> Linux-specific pod security context privileges such as SELinux, AppArmor, Seccomp, or capabilities (POSIX capabilities), and others are not supported.

Reviewer: ```suggestion
Linux-specific pod security context mechanisms (such as SELinux, AppArmor, Seccomp, or custom
POSIX capabilities) are not supported on Windows nodes.
```

Privileged containers are [not supported](#compatibility-v1-pod-spec-containers-securitycontext) on Windows.

> Instead [HostProcess containers](/docs/tasks/configure-pod-container/create-hostprocess-pod) can be used on Windows to perform many of the tasks performed by privileged containers on Linux.

Reviewer: I think this belongs elsewhere, perhaps as a note in https://kubernetes.io/docs/concepts/workloads/pods/#privileged-mode-for-containers ?

Maybe at least restate it there.

Author: Hmm, this is already stated at https://kubernetes.io/docs/concepts/workloads/pods/#privileged-mode-for-containers

> If your cluster has the WindowsHostProcessContainers feature enabled, you can create a [Windows HostProcess pod](https://kubernetes.io/docs/tasks/configure-pod-container/create-hostprocess-pod) by setting the windowsOptions.hostProcess flag on the security context of the pod spec. All containers in these pods must run as Windows HostProcess containers. HostProcess pods run directly on the host and can also be used to perform administrative tasks as is done with Linux privileged containers.

Should we just remove mention of it here?

Reviewer: It's a little bit tidier to use the same or very similar wording in both places; it's also fine to leave them different and then after this merges log a good-first-issue about making the two places match.
(We like to have a nice pool of easily tractable fixes).

Author: I think this would qualify as an easily tractable fix so I can open that issue and add the help-wanted tag after this merges if that could help new contributors get started making PRs into this repo.

---
reviewers:
- jayunit100
- jsturtevant
- marosset
- perithompson
title:    Security For Windows Nodes
content_type: concept
weight: 75
---

<!-- overview -->

This page describes security considerations and best practices specific to the Windows operating system.

<!-- body -->
## Protection for Secret data on nodes

On Windows, data from Secrets are written out in clear text onto the node's local
storage (as compared to using tmpfs / in-memory filesystems on Linux). As a cluster
operator, you should take both of the following additional measures:

1. Use file ACLs to secure the Secrets' file location.
1. Apply volume-level encryption using [BitLocker](https://docs.microsoft.com/windows/security/information-protection/bitlocker/bitlocker-how-to-deploy-on-windows-server).

> ## Container Users

Reviewer: nit:
```suggestion
## Container users
```

[RunAsUsername](/docs/tasks/configure-pod-container/configure-runasusername)
can be specified for Windows Pods or containers to execute the container
processes as specific user. This is roughly equivalent to
[RunAsUser](/docs/concepts/policy/pod-security-policy/#users-and-groups).

Windows containers offer two default user accounts, ContainerUser and ContainerAdministrator.

> The differences between these accounts are covered [here](https://docs.microsoft.com/virtualization/windowscontainers/manage-containers/container-security#when-to-use-containeradmin-and-containeruser-user-accounts).

Reviewer: ```suggestion
The differences between these two user accounts are covered in
[When to use ContainerAdmin and ContainerUser user accounts](https://docs.microsoft.com/virtualization/windowscontainers/manage-containers/container-security#when-to-use-containeradmin-and-containeruser-user-accounts) within Microsoft's _Secure Windows containers_ documentation.
```

Local users can be added to container images during the container build process.

{{< note >}}

> * NanoServer based images run as `ContainerUser` by default

Reviewer: Maybe link to https://docs.microsoft.com/en-us/windows-server/get-started/getting-started-with-nano-server?

Author: This is different than the nanoserver container images.
i will find an appropriate link tho!

Author: Updated

> * ServerCore based images run as `ContainerAdministrator` by default

Reviewer: It's “Server Core” isn't it? I'd hyperlink to https://docs.microsoft.com/en-us/windows-server/administration/server-core/what-is-server-core

Author: That server core is different from the server core container image.
That one topic describes an installation type for the host OS intended for headless server installations.

I will link to https://capz.sigs.k8s.io/developers/development.html#conformance-testing instead tho!

Author: Added a link

{{< /note >}}

> Windows containers can also run as Active-Directory identities by utilizing [Group Managed Service Accounts](docs/tasks/configure-pod-container/configure-gmsa/).

Reviewer: ```suggestion
Windows containers can also run as Active Directory identities by utilizing [Group Managed Service Accounts](/docs/tasks/configure-pod-container/configure-gmsa/)
```

---
reviewers:
- jayunit100
- jsturtevant
- marosset

> - perithompson

Author: TODO: s/perithompson/aravindh
